internal/api/signup.go

```diff
@@ -95,15 +98,26 @@ func (params *SignupParams) ToUserModel(isSSOUser bool) (user *models.User, err
 
 	user.Identities = make([]models.Identity, 0)
 
-	// TODO: Deprecate "provider" field
-	user.AppMetaData["provider"] = params.Provider
+	if params.Provider != "anonymous" {
+		// TODO: Deprecate "provider" field
+		user.AppMetaData["provider"] = params.Provider
 
-	user.AppMetaData["providers"] = []string{params.Provider}
-	if params.Password == "" {
-		user.EncryptedPassword = ""
+		user.AppMetaData["providers"] = []string{params.Provider}
 	}
 
-	return
+	return user, nil
+}
+
+func retrieveSignupParams(r *http.Request) (*SignupParams, error) {
+	params := &SignupParams{}
+	body, err := getBodyBytes(r)
+	if err != nil {
+		return nil, badRequestError("Could not read body").WithInternalError(err)
```

Comment:
This should be:

```suggestion
		return nil, internalServerError("Could not read body").WithInternalError(err)
```

A server like GoTrue will _always_ be able to read the body (because the reverse proxy will only send a full body to it). If it is not able to read the body, it means that something serious is broken (like the network cable being cut in the middle of receiving the body from the proxy). This is not an error that the end user can handle.

Reply:
hmm that reasoning makes sense only if you're using the client library though - if you're hitting the API directly and you don't send a request body, this error should be returned since it's a client error